packages/queries/src/query.ts

```diff
@@ -325,19 +325,71 @@ export class QueryBuilder<
     }
 }
 
-export function query<DB>() {
+export function query<DB>(schema: any) {
     return {
         from<TTable extends Table<DB>>(table: TTable) {
+            type TKeys = Array<Column<DB, TTable>>;
```

Comment:
This whole blob of code is way too dirty.

What you should do is create a function that given a schema returns all the columns 

e.g.

`getColumns(schema: Schema)`

This will make it easier to read, but more importantly to write unit tests for it.